airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -152,25 +155,27 @@ def execute(self, context: 'Context') -> str:
                 Tags=self.tags,
             )
             create_response = json.dumps(create_instance_snap, default=str)
-            self._await_status(
-                'instance_snapshot',
-                self.db_snapshot_identifier,
-                wait_statuses=['creating'],
-                ok_statuses=['available'],
-            )
+            if self.wait_for_completion:
+                self._await_status(
+                    'instance_snapshot',
+                    self.db_snapshot_identifier,
+                    wait_statuses=['creating'],
+                    ok_statuses=['available'],
+                )
         else:
             create_cluster_snap = self.hook.conn.create_db_cluster_snapshot(
                 DBClusterIdentifier=self.db_identifier,
                 DBClusterSnapshotIdentifier=self.db_snapshot_identifier,
                 Tags=self.tags,
             )
             create_response = json.dumps(create_cluster_snap, default=str)
-            self._await_status(
-                'cluster_snapshot',
-                self.db_snapshot_identifier,
-                wait_statuses=['creating'],
-                ok_statuses=['available'],
-            )
+            if self.wait_for_completion:
```

Comment:
Instead of duplicating may be move this outside if [to line 179]?

Reply:
Totally fair.   I'll make the change in both `RdsCreateDbSnapshotOperator` and `RdsCopyDbSnapshotOperator`